simapp/abci.go

```diff
@@ -75,10 +187,56 @@ func (h *VoteExtensionHandler) VerifyVoteExtension() sdk.VerifyVoteExtensionHand
 		case !bytes.Equal(req.Hash, ve.Hash):
 			return &abci.ResponseVerifyVoteExtension{Status: abci.ResponseVerifyVoteExtension_REJECT}, nil
 
-		case len(ve.Data) != 1024:
-			return &abci.ResponseVerifyVoteExtension{Status: abci.ResponseVerifyVoteExtension_REJECT}, nil
 		}
-
 		return &abci.ResponseVerifyVoteExtension{Status: abci.ResponseVerifyVoteExtension_ACCEPT}, nil
 	}
 }
+
+// PreFinalizeBlock -> Extract VEs from previous block and persist in DB.
+func (app *SimApp) NewPreBlockHookHandler() sdk.PreBlocker {
+	return func(ctx sdk.Context, req *abci.RequestFinalizeBlock) (*sdk.ResponsePreBlock, error) {
+		if req.Height > 5 {
+			var ve VoteExtension
+			var eventData *stakingtypes.EventRecord
+			var extVoteInfo []abci.ExtendedVoteInfo
+
+			// Extract ExtendedVoteInfo from txs (encoded at the beginning)
+			bz := req.Txs[0]
+			if err := json.Unmarshal(bz, &extVoteInfo); err != nil {
+				fmt.Println("Error occurred while decoding ExtendedVoteInfo: %w", err)
+			}
+
+			for _, vote := range extVoteInfo {
+				err := json.Unmarshal(vote.VoteExtension, &ve)
+				if err != nil {
+					fmt.Println("Error Occurred while unmarshaling in PreFinalize: ", err)
+					continue
+				}
+
+				if ve.Result == Vote_YES {
```

Comment:
A single YES is enough to persist the VE/execute the SideTx. Should it instead tally the votes to decide on executing or not?

Reply:
Yes we should , similar to how we do in Heimdall : https://github.com/maticnetwork/heimdall/blob/5490d73aaa33bd83f49897eed70a2092b17ac185/app/side_tx_processor.go#L41
cc: @VAIBHAVJINDAL3012